dao/MigrateDAO.php

```diff
@@ -261,4 +261,50 @@ function getReportSID($lid, $sid) {
         return $this->PDOX->rowDie($query, array(':linkId' => $lid, ':siteId' => $sid));
     }
 
+    function getSingleSites() {
+        $query = "SELECT `B`.*
+            FROM {$this->p}migration `A`
+            inner join {$this->p}migration_site `B` on `B`.link_id = `A`.link_id and `B`.state <> 'admin'
+            where `A`.is_admin = 0 order by title, `A`.created_at asc;";
+
+        $arr = array();
+        return $this->PDOX->allRowsDie($query, $arr);
+    }
+
+    function getSingleSitesByState($link_id, $state, $offset, $records_per_page) {
+        if($state == 'all') {
+            $query = "SELECT ifnull(`B`.started_at, 'No start date') as started_at, `B`.site_id, `B`.imported_site_id, `B`.report_url, `B`.state, ifnull(`B`.title, 'No Title') as title
+            FROM {$this->p}migration `A`
+            inner join {$this->p}migration_site `B` on `B`.link_id = `A`.link_id
+            where `A`.is_admin = 0 order by title, `A`.created_at ASC LIMIT $offset, $records_per_page;";
```

Comment:
Always use replaceable parameters rather than construct sql queries using string replacement. Vector for SQL injection otherwise.

Reply:
Added settype($offset, 'integer') to offset and limit